### ISSUE TITLE
Tensorflow 2.4.1 AUC bug

### ISSUE THREAD

**Issue by adalseno** [tensorflow/tensorflow#47132]
Hi, I think there is a bug in tf 2.4.1. If I use tf 2.3.2 it works fine, if I clone the environment and update tf to 2.4.1 I get the following behaviour and auc and val_auc are always at 0.5 no matter how many epochs I run (with a simple dense model; also the loss seems to be erratic):
Epoch 1/10
585/585 [==============================] - 2s 3ms/step - loss: 0.6466 - auc: 0.5065 - val_loss: 0.5479 - val_auc: 0.5000
Epoch 2/10
585/585 [==============================] - 1s 1ms/step - loss: 0.7262 - auc: 0.5000 - val_loss: 0.4576 - val_auc: 0.5000
Epoch 3/10
585/585 [==============================] - 1s 1ms/step - loss: 0.4614 - auc: 0.5000 - val_loss: 0.4271 - val_auc: 0.5000
Epoch 4/10
585/585 [==============================] - 1s 1ms/step - loss: 0.4601 - auc: 0.5000 - val_loss: 0.5069 - val_auc: 0.5000
Epoch 5/10
585/585 [==============================] - 1s 1ms/step - loss: 0.4871 - auc: 0.5000 - val_loss: 0.4679 - val_auc: 0.5000
Epoch 6/10
585/585 [==============================] - 1s 1ms/step - loss: 0.4748 - auc: 0.5000 - val_loss: 0.4341 - val_auc: 0.5000
Epoch 7/10
585/585 [==============================] - 1s 2ms/step - loss: 0.4763 - auc: 0.5000 - val_loss: 0.4290 - val_auc: 0.5000
Epoch 8/10
585/585 [==============================] - 1s 1ms/step - loss: 0.4932 - auc: 0.5000 - val_loss: 0.4367 - val_auc: 0.5000
Epoch 9/10
585/585 [==============================] - 1s 1ms/step - loss: 0.5291 - auc: 0.5000 - val_loss: 0.4290 - val_auc: 0.5000

------------------------

I'm on a Mac Os 10.14.6 and python 3.8.5. (everything else is the same; I cloned the env and copied the notebook)
With tf 2.3.2 auc and val_auc get updated correctly and loss is much better 
Epoch 1/10
585/585 [==============================] - 1s 1ms/step - loss: 0.4133 - auc: 0.6771 - val_loss: 0.3942 - val_auc: 0.7115
Epoch 2/10
585/585 [==============================] - 1s 1ms/step - loss: 0.3959 - auc: 0.6943 - val_loss: 0.3858 - val_auc: 0.7069
Epoch 3/10
585/585 [==============================] - 1s 1ms/step - loss: 0.3889 - auc: 0.6992 - val_loss: 0.3811 - val_auc: 0.6996

**Comment by Saduf2019** [comment]
@adalseno 
Please share simple indented stand code for us to replicate the issue reported or a colab gist with the code and error to analyse.

**Comment by adalseno** [comment]
Here you are the code:
`from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, Activation

model = Sequential()
model.add(Dense(128, activation='relu'))
model.add(Dropout(0.1))
model.add(Dense(2,activation='sigmoid'))
model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['AUC'] )

from keras.callbacks import Callback, EarlyStopping, ReduceLROnPlateau, ModelCheckpoint
# Set early stopping
es = EarlyStopping(monitor = 'val_auc', min_delta = 1e-4, patience = 30, mode = 'max', 
                        baseline = None, restore_best_weights = True, verbose = 1)

rlr = ReduceLROnPlateau(monitor = 'val_auc', factor = 0.1, patience = 25, verbose = 1, 
                            min_delta = 1e-4, mode = 'max', min_lr = 0.00001)



model.fit(x=X_train, 
          y=y_train.values, 
          epochs=200,
          validation_data=(X_test, y_test.values), 
          verbose=1,
          callbacks=[es,rlr]
          )
`
Please note that the "bug" is present even without callbacks (so they are not the problem)

**Comment by adalseno** [comment]
In case you'll need the full code, here you are the first part:
`
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
#import pandas_profiling as pf
%matplotlib inline
#Data from https://www.drivendata.org/competitions/66/flu-shot-learning/data/

train = pd.read_csv('./training_set_features.csv', index_col='respondent_id')
test = pd.read_csv('./test_set_features.csv', index_col='respondent_id')
labels = pd.read_csv('./training_set_labels.csv', index_col='respondent_id')

num_cols = train.select_dtypes('number').columns
cat_cols = ['race', 'sex', 
       'marital_status', 'rent_or_own',  
       'census_msa',  'employment_occupation',
           'age_group', 'education',  'income_poverty',
        'employment_status']

from sklearn.pipeline import Pipeline
from sklearn.preprocessing import MinMaxScaler, StandardScaler, OneHotEncoder
from sklearn.impute import SimpleImputer
from sklearn.compose import ColumnTransformer
from category_encoders import OrdinalEncoder as oe

num_pipe= Pipeline([
    ('impute', SimpleImputer(strategy='constant', fill_value = -1  )),
    ('scale', MinMaxScaler())
])

cat_pipe = Pipeline([
        ('impute', SimpleImputer(strategy='constant', fill_value = 'missing'  )),
        ('encode', OneHotEncoder(handle_unknown='ignore'))

])

ct = ColumnTransformer(
    [('numerical', num_pipe, num_cols), 
    ('onehot', cat_pipe, cat_cols)],
    remainder='drop')

sel_cols = train.columns

for col in num_cols:
    train[col] = train[col].fillna(value=-1)

for col in (cat_cols):
    train[col] = train[col].fillna(value='missing')

train = pd.DataFrame(train, columns=sel_cols)

train = ct.fit_transform(train)

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split( train, labels, test_size=0.3, random_state=68)


`

**Comment by Saduf2019** [comment]
@adalseno 
I ran the code shared and face a different issue, please find [gist here](https://colab.research.google.com/gist/Saduf2019/a2520a6f2a5591931ffff8112ad13713/untitled535.ipynb) and share all dependencies such that we can replicate the issue faced or if possible share a colab gist with the error.

**Comment by adalseno** [comment]
I apologize for the late reply. The issue was with a missing package. I solved adding:
`!pip install category_encoders`
It works fine now and the issue is there:

Epoch 1/200
585/585 [==============================] - 2s 3ms/step - loss: 0.6616 - auc: 0.5036 - val_loss: 0.5443 - val_auc: 0.5000
Epoch 2/200
585/585 [==============================] - 1s 2ms/step - loss: 2.3236 - auc: 0.5000 - val_loss: 1.0065 - val_auc: 0.5000
Epoch 3/200
585/585 [==============================] - 1s 2ms/step - loss: 2.9277 - auc: 0.5000 - val_loss: 1.4446 - val_auc: 0.5000
Epoch 4/200
585/585 [==============================] - 1s 2ms/step - loss: 2.2756 - auc: 0.5000 - val_loss: 0.5043 - val_auc: 0.5000

**Comment by Saduf2019** [comment]
@adalseno 
Thank you for your update, please move this issue to closed status as it is resolved.
